skbio/stats/composition.py

```diff
@@ -608,10 +610,271 @@ def centralize(mat):
 
     """
     mat = closure(mat)
-    cen = ss.gmean(mat, axis=0)
+    cen = scipy.stats.gmean(mat, axis=0)
     return perturb_inv(mat, cen)
 
 
+@experimental(as_of="0.4.0")
+def ancom(mat, cats,
+          alpha=0.05,
+          multicorr=False,
+          tau=0.02,
+          theta=0.1,
+          func=scipy.stats.ttest_ind):
+    r"""
+    Calculates pairwise log ratios between all otus
+    and performs a signficance test to determine if there is a
+    significant difference in feature ratios with respect to the
+    variable of interest
+
+    In an experiment with only two treatments, this test tests the
+    following hypothesis for feature :math:`i`
+    :math:`H_{0i}:\E[\ln(u_i^{(1)})] = \E[\ln(u_i^{(2)})]`
+
+    where :math:`u_i^{(1)}` is the mean abundance for feature
+    :math:`i` in the first group and :math:`u_i^{(2)}` is the
+    mean abundance for feature :math:`i` in the second group.
+
+    This method can be extended two an arbitrary number of classes
+    by passing in a different statistical function.
+
+
+    Parameters
+    ----------
+    mat: pd.DataFrame or array_like
+       A 2D matrix where
+       rows = samples
+       columns = features
+    cat: pd.Series or array_like
+       Vector of categories
+    alpha : float
+       Significance level for each of the statistical tests
+    multicorr: bool
+       Runs multiple comparisons correction or not.
+       If specified, this will run Holm-Boniferroni
+       correction
+    tau: float
+       A constant used to determine an appropriate
+       cutoff (default=0.02)
+    theta : float
+       Lower bound for the proportion of differiential features
+       (default=0.1)
+    func: function
+       A statistical signficance function to test for
+       signficance between classes.
+       The default is scipy.stats.ttest_ind
+
+    Returns:
+    --------
+    W : pd.Series
+        List of W statistics
+    reject : pd.Series
+        Indicates if the null hypothesis has been rejected
+
+    References
+    ----------
+    ..[1] S. Mandal, 'Analysis of composition of microbiomes:
+          a novel method for studying microbial composition'
```

Comment:
Probably should do that.  There are  other references in `composition.py` don't have journal, issue, date that should also be fixed.